Apply reviewer feedback to Security section in agent mode docs (#8710)
microsoft/vscode-docs: docs/copilot/chat/chat-agent-mode.md

In case you want to auto-approve _all_ tools and terminal commands, you can now use the experimental `setting(chat.tools.autoApprove)` setting. This will automatically approve all tool and command invocations, and VS Code will not ask for confirmation when a language model wishes to run tools.

> [!CAUTION]

> > With this setting enabled, you don't have the opportunity to cancel potentially destructive actions a model wants to take. This significantly reduces the security protections that agent mode provides. Only enable this setting if you fully understand the security implications and are working in a trusted, isolated environment.

Reviewer: ```suggestion
> With this setting enabled, you don't have the opportunity to cancel potentially destructive actions a model wants to take. This significantly reduces the security protections in agent mode. Only enable this setting if you fully understand the security implications and are working in a trusted, isolated environment.
```

#### Permission-based security architecture

Agent mode uses a permission-based security model where you maintain control over potentially risky operations:

**Approve tool calls and terminal commands**: Before running tools and terminal commands, agent mode requests explicit confirmation. You can approve actions for the current session, workspace, or all future invocations. See [Manage tool approvals](#manage-tool-approvals) for details.

**Review file changes**: While agent mode can automatically write files, you can review proposed changes in the chat interface before they are applied.

**Tool enablement**: You can selectively enable or disable specific tools using the tools picker, giving you precise control over what capabilities are available.

#### MCP security

Before starting an MCP server, you need to trust it explicitly. This prevents malicious code execution on startup and shows a trust prompt when installing or when MCP server configuration changes.

#### Additional safeguards

> **Secure credentials store**: VS Code uses a secure credentials store for tool configuration, helping protect sensitive authentication information.

Reviewer 2: ```suggestion
**Secure credentials store**: VS Code uses a secure credentials store for sensitive tool input parameters, helping protect sensitive authentication information.
```

**Request limits**: The `setting(chat.agent.maxRequests)` setting limits the number of requests agent mode can make, preventing runaway operations.

> **Session isolation**: Tool approvals can be scoped to the current session, allowing you to grant temporary permissions that don't persist.

Reviewer 2: ```suggestion
**Session isolation**: You can scope tool approvals to the current session, enabling you to grant temporary permissions that don't persist.
```

### User responsibility and best practices

> While agent mode includes security protections, users should follow these best practices:

Reviewer 2: ```suggestion
While agent mode includes security protections, you should follow these best practices:
```

> 

Reviewer 2: Use a bulleted list for these items

**Review suggested commands before approval**: Carefully examine terminal commands and tool invocations before approving them. Don't approve operations you don't understand.

**Review MCP tool configuration before starting them**: Before enabling MCP servers or extension tools, understand what they can do, what data they can access, and verify the publisher information.

**Review MCP server publisher information**: Check the source and reputation of MCP servers before installing and using them in your environment.

**Verify proposed changes to critical files**: Pay special attention when agent mode wants to modify important files like configuration files, security settings, or build scripts.

> **Consider using dev containers or VMs for isolation**: For enhanced security, run agent mode operations in isolated environments like dev containers or virtual machines to limit potential impact.

Reviewer 2: ```suggestion
**Consider using dev containers or VMs for isolation**: For enhanced security, run agent mode operations in isolated environments like [dev containers](https://code.visualstudio.com/docs/devcontainers/containers) or virtual machines to limit potential impact.
```
